assets/js/modules/analytics/datastore/accounts.js

```diff
@@ -0,0 +1,209 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import { actions as propertyActions } from './properties';
+import { actions as profileActions } from './profiles';
+
+// Actions
+const FETCH_ACCOUNTS_PROPERTIES_PROFILES = 'FETCH_ACCOUNTS_PROPERTIES_PROFILES';
+const RECEIVE_ACCOUNTS = 'RECEIVE_ACCOUNTS';
+const RECEIVE_ACCOUNTS_PROPERTIES_PROFILES_COMPLETED = 'RECEIVE_ACCOUNTS_PROPERTIES_PROFILES_COMPLETED';
+const RECEIVE_ACCOUNTS_PROPERTIES_PROFILES_FAILED = 'RECEIVE_ACCOUNTS_PROPERTIES_PROFILES_FAILED';
+
+export const INITIAL_STATE = {
+	accounts: undefined,
+	isFetchingAccountsPropertiesProfiles: false,
+};
+
+export const actions = {
+	*fetchAccountsPropertiesProfiles() {
+		return {
+			payload: {},
+			type: FETCH_ACCOUNTS_PROPERTIES_PROFILES,
+		};
+	},
+
+	/**
+	 * Creates an action for receiving accounts.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {Array} accounts Accounts to receive.
+	 * @return {Object} action object.
+	 */
+	receiveAccounts( accounts ) {
+		invariant( Array.isArray( accounts ), 'accounts must be an array.' );
+
+		return {
+			payload: { accounts },
+			type: RECEIVE_ACCOUNTS,
+		};
+	},
+
+	receiveAccountsPropertiesProfilesCompleted() {
+		return {
+			payload: {},
+			type: RECEIVE_ACCOUNTS_PROPERTIES_PROFILES_COMPLETED,
+		};
+	},
+
+	receiveAccountsPropertiesProfilesFailed( error ) {
+		invariant( error, 'error is required.' );
+
+		return {
+			payload: { error },
+			type: RECEIVE_ACCOUNTS_PROPERTIES_PROFILES_FAILED,
+		};
+	},
+};
+
+export const controls = {
+	[ FETCH_ACCOUNTS_PROPERTIES_PROFILES ]: () => {
+		return API.get( 'modules', 'analytics', 'accounts-properties-profiles' );
+	},
+};
+
+export const reducer = ( state, { type, payload } ) => {
+	switch ( type ) {
+		case FETCH_ACCOUNTS_PROPERTIES_PROFILES: {
+			return {
+				...state,
+				isFetchingAccountsPropertiesProfiles: true,
+			};
+		}
+
+		case RECEIVE_ACCOUNTS: {
+			const { accounts } = payload;
+
+			return {
+				...state,
+				accounts: [ ...accounts ],
+			};
+		}
+
+		case RECEIVE_ACCOUNTS_PROPERTIES_PROFILES_COMPLETED: {
+			return {
+				...state,
+				isFetchingAccountsPropertiesProfiles: false,
+			};
+		}
+
+		case RECEIVE_ACCOUNTS_PROPERTIES_PROFILES_FAILED: {
+			const { error } = payload;
+
+			return {
+				...state,
+				error,
+				isFetchingAccountsPropertiesProfiles: false,
+			};
+		}
+
+		default: {
+			return { ...state };
+		}
+	}
+};
+
+export const resolvers = {
+	*getAccounts() {
+		try {
+			const response = yield actions.fetchAccountsPropertiesProfiles();
+			const { accounts, properties, profiles, matchedProperty } = response;
```

Comment:
If `matchedProperty` is not in the response, this destructuring will still work right (not sure about JS here)? I.e. this would just be `undefined` correct?

Reply:
>  I.e. this would just be undefined correct?

Yes, correct 👍

Reply:
Yup! This will just be `undefined`, which I think is fine here.